src/meltano/core/utils/__init__.py

```diff
@@ -366,7 +366,23 @@ def set_at_path(d, path, value):
     final[tail] = value
 
 
-def expand_env_vars(raw_value, env: Dict):
+class EnvironmentVariableNotSetError(Exception):
+    """Occurs when a referenced environment variable is not set."""
+
+    def __init__(self, env_var: str):
+        """Initialize the error.
+        Args:
+            env_var: the unset environment variable name
+        """
+        super().__init__(env_var)
+        self.env_var = env_var
+
+    def __str__(self) -> str:
+        """Return the error as a string."""
+        return f"{self.env_var} referenced but not set."
+
+
+def expand_env_vars(raw_value, env: Dict, raise_error: bool = False):
```

Comment:
Wdyt about calling the new keyword arg `raise_if_missing`?

Reply:
Yeah I think that's more descriptive and clear. Updated.